### PR TITLE
DH backend interfaces

### PR DESCRIPTION
**Pull request by public**
DHBackend and docs.

### CODE REVIEW

File: docs/hazmat/backends/interfaces.rst [hunk 1]
```diff
@@ -487,3 +487,95 @@ A specific ``backend`` may provide one or more of these interfaces.
...
+    .. versionadded:: 0.8
+
+    A backend with methods for doing Diffie-Hellman key exchange.
+
+    .. method:: generate_dh_parameters(key_size)
+
+        :param int key_size: The bit length of the prime modulus to generate.
+
+        :return: A new instance of a
+            :class:`~cryptography.hazmat.primitives.interfaces.DHParameters`
```
Comment (reaperhulk) [inline docs/hazmat/backends/interfaces.rst]: Wrong module path

Comment (reaperhulk) [inline docs/hazmat/backends/interfaces.rst]: Err, it's actually not right now but... *leaves a comment saying we should move those interfaces in the other PR*

File: docs/hazmat/backends/interfaces.rst [hunk 2]
```diff
@@ -487,3 +487,95 @@ A specific ``backend`` may provide one or more of these interfaces.
...
+        :return: A new instance of a
+            :class:`~cryptography.hazmat.primitives.interfaces.DHPrivateKey`
+            provider.
+
+        :raises ValueError: If ``key_size`` is not at least 512.
+
+    .. method:: load_dh_private_numbers(numbers)
+
+        :param numbers: A
+            :class:`~cryptography.hazmat.primitives.dh.DHPrivateNumbers`
```
Comment (reaperhulk) [inline docs/hazmat/backends/interfaces.rst]: This, however, is the wrong module path (several others need fixing as well)

**Comment by alex** [comment]
docs refer to a class that has been moved

**Comment by public** [comment]
Rebased.

**Pull request closed without merging**